controllers/gateway/operator.go

```diff
@@ -159,22 +168,13 @@ func (goc *gwOperationCtx) operate() error {
 		_, err = goc.controller.kubeClientset.AppsV1().Deployments(goc.gw.Namespace).Create(gatewayDeployment)
 		if err != nil {
 			goc.log.Error().Err(err).Msg("failed gateway deployment")
-			goc.gw.Status = v1alpha1.NodePhaseError
+			goc.markGatewayPhase(v1alpha1.NodePhaseError, "failed gateway deployment")
 		} else {
-			goc.gw.Status = v1alpha1.NodePhaseRunning
+			// expose gateway if service is configured
 			if goc.gw.Spec.Service.Port != 0 {
 				goc.createGatewayService()
```

Comment:
why do we fail silently here?

Reply:
As the service is not absolutely critical part of the gateway. An user can create the service manually in the event of service creation failure.